### PR TITLE
Added offset prop to Tooltip and Popover components

### PR DESCRIPTION
**Pull request by illiteratewriter**

### CODE REVIEW

**Comment by illiteratewriter** [comment]
I am sorry, but I have no clue what I did. Still figuring out git. I was trying to push the `tests` and `docs` and i tried to change the commit message. I'll do the docs and test again and do a PR.

**Comment by illiteratewriter** [comment]
I figured it out. and have added `tests` and updated the `docs`

File: src/__tests__/Popover.spec.js [hunk 1]
```diff
@@ -513,5 +513,17 @@ describe('Popover', () => {
 
       wrapper.detach();
     });
+
+    it('should pass down offset', ()=> {
```
Comment (gergely-nagy) [inline src/__tests__/Popover.spec.js]: Missing space before =>

File: src/__tests__/Popover.spec.js [hunk 2]
```diff
@@ -513,5 +513,17 @@ describe('Popover', () => {
...
+      const wrapper = mount(
+        <Popover
+          isOpen
+          target="innerTarget"
+          offset='100'>Popover content</Popover>
```
Comment (gergely-nagy) [inline src/__tests__/Popover.spec.js]: - Unexpected usage of singlequote , use " " instead
- The closing bracket must be aligned with the line containing the opening tag

File: src/__tests__/Popover.spec.js [hunk 3]
```diff
@@ -513,5 +513,17 @@ describe('Popover', () => {
...
+      );
+
+      expect(wrapper.find(PopperContent).props().offset).toEqual('100')
```
Comment (gergely-nagy) [inline src/__tests__/Popover.spec.js]: Missing semicolon

File: src/__tests__/Tooltip.spec.js [hunk 4]
```diff
@@ -236,6 +236,18 @@ describe('Tooltip', () => {
     wrapper.unmount();
   });
 
+  it('should pass down offset', ()=> {
```
Comment (gergely-nagy) [inline src/__tests__/Tooltip.spec.js]: Missing space before =>

File: src/__tests__/Tooltip.spec.js [hunk 5]
```diff
@@ -236,6 +236,18 @@ describe('Tooltip', () => {
...
+    const wrapper = mount(
+      <Tooltip
+        isOpen
+        target="target"
+        offset='100'>Tooltip content</Tooltip>
```
Comment (gergely-nagy) [inline src/__tests__/Tooltip.spec.js]: - Unexpected usage of singlequote , use " " instead
- The closing bracket must be aligned with the line containing the opening tag

File: src/__tests__/Tooltip.spec.js [hunk 6]
```diff
@@ -236,6 +236,18 @@ describe('Tooltip', () => {
...
+    );
+
+    expect(wrapper.find(PopperContent).props().offset).toEqual('100')
```
Comment (gergely-nagy) [inline src/__tests__/Tooltip.spec.js]: Missing semicolon

**Comment by TheSharpieOne** [comment]
Props (within JSX) should use double quotes (just as HTML attributes would be) everything else should be single quotes.

**Comment by gergely-nagy** [comment]
@illiteratewriter we use [Travis-CI](https://travis-ci.org/) to automatically run our unit tests and for code linting.

When a pull request is opened on GitHub (or commits are added to the pull request), Travis CI receives a notification and runs a build. 

During the build, Travis CI update the status icon of the pull request to one of the following statuses:

- a warning that the build is still running.
- a notification that the build failed – the pull request should not be merged.
- a notification that the build succeeded – the pull request can be merged.

Your PR fails, because [eslint](https://eslint.org/) found some error. You can view the log [here](https://travis-ci.org/reactstrap/reactstrap/builds/378883831).

You should run linter before commits.
To run ESLint all you need to do is run `npm run lint` from the inside of your project folder. 

I hope this helps you. :smiley:

**Comment by illiteratewriter** [comment]
Thank you for being patient and helping me with this. It was great working with you. :)

**Pull request merged**